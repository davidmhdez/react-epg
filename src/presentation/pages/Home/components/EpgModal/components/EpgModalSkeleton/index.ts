export {default} from "./EpgModalSkeleton.tsx"