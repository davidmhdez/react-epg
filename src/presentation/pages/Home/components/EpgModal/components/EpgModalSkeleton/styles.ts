import styled from "styled-components";

export const EpgModalSkeletonWrapper = styled.div`
    width: 100%;
`

export const EpgModalSkeletonTable = styled.div`
    overflow: hidden;
    width: 100%;
`