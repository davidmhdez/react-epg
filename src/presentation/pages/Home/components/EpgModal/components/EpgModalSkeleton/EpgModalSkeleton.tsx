import {EpgModalSkeletonTable, EpgModalSkeletonWrapper} from "./styles.ts";
import {SkeletonTheme} from "react-loading-skeleton";
import EpgTableHeaderSkeleton from "./components/EpgTableHeaderSkeleton";
import EpgTableRowSkeleton from "./components/EpgTableRowSkeleton";

const EpgModalSkeleton = () => {
    return (
        <EpgModalSkeletonWrapper>
            <SkeletonTheme baseColor="#6C6C6CAD" highlightColor="#7E7E7EAD">
                <EpgModalSkeletonTable>
                    <EpgTableHeaderSkeleton/>
                    <EpgTableRowSkeleton programColsWidth={[150, 250, 140, 350, 260]}/>
                    <EpgTableRowSkeleton programColsWidth={[250, 180, 220, 250, 560]}/>
                    <EpgTableRowSkeleton programColsWidth={[330, 190, 170, 520, 90]}/>
                </EpgModalSkeletonTable>
            </SkeletonTheme>
        </EpgModalSkeletonWrapper>
    );
};

export default EpgModalSkeleton;
