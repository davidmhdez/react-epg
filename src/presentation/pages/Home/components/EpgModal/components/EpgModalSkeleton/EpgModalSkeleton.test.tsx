import {render} from "@testing-library/react";
import EpgModalSkeleton from "./EpgModalSkeleton.tsx";

describe('EpgModalSkeleton test', () => {
    test('should match with snapshot', () => {
        const { container } = render(<EpgModalSkeleton/>)
        expect(container).toMatchSnapshot()
    })
})