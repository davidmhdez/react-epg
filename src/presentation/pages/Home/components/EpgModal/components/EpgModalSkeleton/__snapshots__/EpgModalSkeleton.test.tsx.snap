// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpgModalSkeleton test should match with snapshot 1`] = `
<div>
  <div
    class="sc-aYaIB bSTRcM"
    data-testid="epgModalSkeleton"
  >
    <div
      class="sc-gEvDqW jwQatR"
    >
      <div
        class="sc-eqUzNf giyJNu"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton dayText"
            style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton hourText"
            style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton hourText"
            style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton hourText"
            style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton hourText"
            style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton hourText"
            style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton hourText"
            style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div
        class="sc-fqkwJk hQWqGa"
      >
        <div
          class="sc-dcJtft JTiyR"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton channelNumber"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton channelImage"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO eroMrp"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO dVWzxe"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO yurWs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO edhPWr"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO jrLrpN"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
      <div
        class="sc-fqkwJk hQWqGa"
      >
        <div
          class="sc-dcJtft JTiyR"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton channelNumber"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton channelImage"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO dVWzxe"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO cvnIIM"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO jjiyZR"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO dVWzxe"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO hhSMjW"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
      <div
        class="sc-fqkwJk hQWqGa"
      >
        <div
          class="sc-dcJtft JTiyR"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton channelNumber"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton channelImage"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO hmVHgF"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO goiddJ"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO gbPRUz"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO kDunDS"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO buXabQ"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programTitle"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton programDuration"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
